controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1515,3 +1420,153 @@ def test_identity_probe(self):
         request = Mock()
         context = Mock()
         self.servicer.Probe(request, context)
+
+
+class TestValidateVolumeCapabilities(BaseControllerSetUp, CommonControllerTest):
+
+    def get_tested_method(self):
+        return self.servicer.ValidateVolumeCapabilities
+
+    def get_tested_method_response_class(self):
+        return csi_pb2.ValidateVolumeCapabilitiesResponse
+
+    def setUp(self):
+        super().setUp()
+
+        arr_type = XIVArrayMediator.array_type
+        self.request.volume_id = "{}:wwn1".format(arr_type)
+        self.request.parameters = {config.PARAMETERS_POOL: "pool1"}
+
+        self.mediator.get_object_by_id = Mock()
+        self.mediator.get_object_by_id.return_value = utils.get_mock_mediator_response_volume(10, "vol", "wwn2", "a9k")
+        self.request.volume_capabilities = [self.volume_capability]
+
+    def _assertResponse(self, response, expected_status_code, expected_details_substring):
+        self.assertEqual(self.context.code, expected_status_code)
+        self.assertTrue(expected_details_substring in self.context.details)

```

Comment:
@oriyarde 
There is a test for the message information that we send upon errors.
In this case, we check the substring of the message.
Checking a substring of a string is weaker than checking the whole string. Thus, I think we should compare the entire string.
WDYT?

Reply:
depends on what is important for us to keep.
if we want to "protect" from every change in the string, including around the substring, then we have two options:
1. import the common const in the test, which you could argue is an even weaker test than having a separate substring
2. duplicate the entire string, forcing the maintainer to update in two places for every change

I don't like the second option much, but I'm ok with all 3 approaches

note that there might be more tests like this throughout the project

Reply:
I prefer option two because it makes the test stronger.
@ArbelNathan, the choice is yours.

Reply:
please resolve after the work on the conversation is done or at least after we decided what to do..

Reply:
if we choose option 2 so the work here is very big (that message test is on almost every test in the project) so it will need a separate ticket.
@zingero if you you prefer that option please open a ticket for it .